endpoint: Add DebugPolicy option

jrajahalme opened the pull request:
Add endpoint DebugPolicy option that, if enabled, logs endpoint policy
map update details to /var/run/cilium/state/endpoint-policy.log.

The new DebugPolicy option is enabled if the new flag
--debug-verbose=policy is set, but can be enabled also independently
via:

  cilium endpoint config <EPID> DebugPolicy=true

This feature was first developed in the v1.7 branch, so no 1.7 backport is needed.

Signed-off-by: Martynas Pumputis <m@lambda.lt>
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

```release-note
Added new Cilium agent option --debug-verbose=policy to log policy map updates.
```

brb committed:
endpoint: Add DebugPolicy option

Add endpoint DebugPolicy option that, if enabled, logs endpoint policy
map update details to /var/run/cilium/state/endpoint-policy.log.

The new DebugPolicy option is enabled if the new flag
--debug-verbose=policy is set, but can be enabled also independently
via:

  cilium endpoint config <EPID> DebugPolicy=true

Signed-off-by: Martynas Pumputis <m@lambda.lt>
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

ianvernon reviewed (approved):
Looks solid - hope you can use this to debug issues more easily in the future :-)

kaworu commented on pkg/endpoint/log.go:
```diff
@@ -114,3 +139,61 @@ func (e *Endpoint) UpdateLogger(fields map[string]interface{}) {
 	atomic.StorePointer(&e.logger, unsafe.Pointer(l))
 }
+
+func (e *Endpoint) updatePolicyLogger(fields map[string]interface{}) {
```
The caller need to hold `Endpoint.mutex` for writing correct? If yes, can we add a comment please?

About `UpdateLogger` comment on holding the mutex, my understanding is that the caller must hold _`Endpoint.mutex`_ (not `Endpoint.Mutex`) for _writing_ (not for reading) even when `fields` is `nil` as we can reach the `atomic.StorePointer` at line 140.

jrajahalme replied:
I'll update the locking comments ;-)

jrajahalme committed:
endpoint: Update lock requirement comments

Endpoint's Mutex has been renamed as 'mutex'. Update comments to
reflect this and also the lock level requirement (Lock for writing,
RLock for reading).

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
